src/shared_realm.hpp

```diff
@@ -328,6 +333,24 @@ class InvalidEncryptionKeyException : public std::logic_error {
 public:
     InvalidEncryptionKeyException() : std::logic_error("Encryption key must be 64 bytes.") {}
 };
+
+// FIXME Those are exposed for Java async queries, mainly because of handover related methods.
+class _impl::RealmFriend {
```

Comment:
The `_impl` namespace is internal to the object store, while this class is explicitly intended to be exposed. Those two don't go together.